### PR TITLE
fix: test(fgs/function): repair some BUGs of the depend_list and support a test to check it

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
- There is no result of parameter ` depend_list` in the API repsonse body, because of the parameter `depend_version_list` is instead of parameter `depend_list`.
- The response list of the parameter `depend_version_list` is in order.
- The UT does not contain the function of the 'obs' code type .

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. replace the parameter 'depend_list' with the parameter 'depend_verison_list'.
2. update the parameter type in the schema, from list to set.
3. support the new test to check the obs type function.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (99.95s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       99.995s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 1]
```diff
@@ -285,13 +299,48 @@ func TestAccFgsV2Function_logConfig(t *testing.T) {
...
+resource "huaweicloud_obs_bucket" "test" {
+  bucket = "%[1]s"
+  acl    = "private"
+
+  provisioner "local-exec" {
+    command = "echo '${var.script_content}' >> ${abspath(path.root)}/test.py\nzip -r ${abspath(path.root)}/test.zip ${abspath(path.root)}/test.py"
+  }
+  provisioner "local-exec" {
+    command = "rm ${abspath(path.root)}/test.zip ${abspath(path.root)}/test.py"
+	when    = destroy
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: Alignment with equal sign

**Comment by Lance52259** [comment]
```
 make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (97.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       97.327s
```

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: fixed.

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 2]
```diff
@@ -285,13 +299,48 @@ func TestAccFgsV2Function_logConfig(t *testing.T) {
...
+  provisioner "local-exec" {
+    command = "rm test.zip test.py"
+    when    = destroy
+  }
+}
+
+resource "huaweicloud_obs_bucket_object" "test" {
+  bucket = huaweicloud_obs_bucket.test.bucket
+  key    = "test.zip"
+  source = "${abspath("./")}/test.zip"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: using abspath directly will be fine

```suggestion
  source = abspath("./test.zip")
```

Comment (Lance52259) [reply huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (104.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       104.943s
```

**Pull request merged**